src/main/java/seedu/duke/parsers/GradeParser.java

```diff
@@ -0,0 +1,38 @@
+package seedu.duke.parsers;
+
+import seedu.duke.commands.Command;
+import seedu.duke.commands.GradeCommand;
+import seedu.duke.exceptions.ModHappyException;
+import seedu.duke.exceptions.ParseException;
+import seedu.duke.util.StringConstants;
+
+import java.util.HashMap;
+import java.util.Objects;
+
+public class GradeParser extends Parser {
+    public static final String MODULE_CODE = StringConstants.MODULE_CODE;
+    public static final String MODULE_GRADE = StringConstants.MODULE_GRADE;
+
+    // Unescaped regex for testing:
+    // (/m\s+(?<moduleCode>\w+)(\s+/g\s+(?<moduleGrade>(?i)([A-B][+-]?|[C-D][+]?|F))))
+    private static final String GRADE_FORMAT = "(/m\\s+(?<moduleCode>\\w+)(\\s+/g\\s+"
+            + "(?<moduleGrade>(?i)([A-B][+-]?|[C-D][+]?|F))))";
```

Comment:
Same as the earlier comment, since the grade parameter is mandatory, we can consider getting rid of the flag.

Also, we should probably support S and U as valid grades as well.

Reply:
Same as above, and I have added support for CS, CU, S and U.